src/ucp/wireup/wireup_cm.c

```diff
@@ -1068,8 +1097,35 @@ void ucp_cm_server_conn_request_cb(uct_listener_h listener, void *arg,
                      UCT_DEVICE_NAME_MAX);
     memcpy(ucp_conn_request->remote_dev_addr, remote_data->dev_addr,
            remote_data->dev_addr_length);
+
+    /* [ sa_data | ucp_address | user_data ]
+     * |      conn_priv_data_length        |
+     * 1. sa_data is fixed of size, so copy it directly */
     memcpy(&ucp_conn_request->sa_data, remote_data->conn_priv_data,
-           remote_data->conn_priv_data_length);
+           sizeof(ucp_conn_request->sa_data));
+
+    offset_iter = remote_data->conn_priv_data;
+    ucs_serialize_next_raw(&offset_iter, const void*,
+                           sizeof(ucp_wireup_sockaddr_data_t));
```

Comment:
instead of memcpy+ucs_serialize_next_raw:
```
ucp_conn_request->sa_data = 
       *ucs_serialize_next(&offset_iter, const ucp_wireup_sockaddr_data_t*);
```